persistence/src/commonTest/kotlin/com/wire/kalium/persistence/dao/UserDAOTest.kt

```diff
@@ -762,6 +763,68 @@ class UserDAOTest : BaseDatabaseTest() {
+            team = "Team",
+            connectionStatus = ConnectionEntity.State.ACCEPTED,
+            previewAssetId = UserAssetIdEntity("PreviewAssetId", "PreviewAssetDomain"),
+            completeAssetId = UserAssetIdEntity("CompleteAssetId", "CompleteAssetDomain"),
+            availabilityStatus = UserAvailabilityStatusEntity.AVAILABLE,
+            userType = UserTypeEntity.STANDARD,
+            botService = BotIdEntity("BotService", "BotServiceDomain"),
+            deleted = false,
+            hasIncompleteMetadata = false,
+            expiresAt = null,
+            defederated = false,
+        )
+        db.userDAO.insertUser(user)
+        val updatedTeamMemberUser = user1.copy(
+            name = "newName",
+            handle = "newHandle",
+            email = "newEmail",
+            phone = "newPhone",
+            accentId = 2,
+            team = "newTeam",
+            connectionStatus = ConnectionEntity.State.PENDING,
+            previewAssetId = UserAssetIdEntity("newPreviewAssetId", "newPreviewAssetDomain"),
+            completeAssetId = UserAssetIdEntity("newCompleteAssetId", "newCompleteAssetDomain"),
+            availabilityStatus = UserAvailabilityStatusEntity.BUSY,
+            userType = UserTypeEntity.EXTERNAL,
+            botService = BotIdEntity("newBotService", "newBotServiceDomain"),
+            deleted = true,
+            hasIncompleteMetadata = true,
+            expiresAt = Clock.System.now(),
+            defederated = true,
+        )
+        db.userDAO.upsertTeamMembers(listOf(updatedTeamMemberUser))
+        val result = db.userDAO.getUserByQualifiedID(user1.id).first()
+        assertTrue {
+            result != null &&
+            result.name == updatedTeamMemberUser.name &&
+            result.handle == updatedTeamMemberUser.handle &&
+            result.email == updatedTeamMemberUser.email &&
+            result.phone == updatedTeamMemberUser.phone &&
+            result.accentId == updatedTeamMemberUser.accentId &&
+            result.team == updatedTeamMemberUser.team &&
+            result.connectionStatus == updatedTeamMemberUser.connectionStatus &&
+            result.previewAssetId == updatedTeamMemberUser.previewAssetId &&
+            result.completeAssetId == updatedTeamMemberUser.completeAssetId &&
+            result.availabilityStatus != updatedTeamMemberUser.availabilityStatus && // should not be updated
+            result.userType != updatedTeamMemberUser.userType && // should not be updated
+            result.botService == updatedTeamMemberUser.botService &&
+            result.deleted != updatedTeamMemberUser.deleted && // should not be updated
+            result.hasIncompleteMetadata != updatedTeamMemberUser.hasIncompleteMetadata && // should not be updated
+            result.expiresAt != updatedTeamMemberUser.expiresAt && // should not be updated
+            result.defederated != updatedTeamMemberUser.defederated // should not be updated
+        }
```

Comment:
When this test breaks, all we'll see is `expected: true, actual:false`.

```suggestion
        assertNotNull(result)
        assertEquals(updatedTeamMemberUser.name, result.name)
        assertEquals(updatedTeamMemberUser.handle, result.handle)
        assertEquals(updatedTeamMemberUser.email, result.email)
        assertEquals(updatedTeamMemberUser.phone, result.phone)
        assertEquals(updatedTeamMemberUser.accentId, result.accentId)
        assertEquals(updatedTeamMemberUser.team, result.team)
        assertEquals(updatedTeamMemberUser.connectionStatus, result.connectionStatus)
        assertEquals(updatedTeamMemberUser.previewAssetId, result.previewAssetId)
        assertEquals(updatedTeamMemberUser.completeAssetId, result.completeAssetId)
        assertNotEquals(updatedTeamMemberUser.availabilityStatus, result.availabilityStatus)
        assertNotEquals(updatedTeamMemberUser.userType, result.userType)
        assertNotEquals(updatedTeamMemberUser.botService, result.botService)
        assertNotEquals(updatedTeamMemberUser.deleted, result.deleted)
        assertNotEquals(updatedTeamMemberUser.hasIncompleteMetadata, result.hasIncompleteMetadata)
        assertNotEquals(updatedTeamMemberUser.expiresAt, result.expiresAt)
        assertNotEquals(updatedTeamMemberUser.defederated, result.defederated)
```

It's going to be easier to find what is going on.